src/HDF5.jl

```diff
@@ -589,22 +590,27 @@ See also
 """
 function create_dataset(
     parent::Union{File,Group},
-    path::AbstractString,
+    path::Union{AbstractString, Nothing},
     dtype::Datatype,
     dspace::Dataspace;
     dcpl::DatasetCreateProperties = DatasetCreateProperties(),
     dxpl::DatasetTransferProperties = DatasetTransferProperties(),
     dapl::DatasetAccessProperties = DatasetAccessProperties(),
     pv...
 )
-    haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
+    path !== nothing && haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
     pv = setproperties!(dcpl,dxpl,dapl; pv...)
     isempty(pv) || error("invalid keyword options")
-    Dataset(API.h5d_create(parent, path, dtype, dspace, _link_properties(path), dcpl, dapl), file(parent), dxpl)
+    if path isa Nothing
```

Comment:
```suggestion
    if path === nothing
```

Perhaps this is clearer? I'm not sure what the practical difference is.

Reply:
`isnothing(path)`

Reply:
`isnothing` does not exist in older Julia whereas `===` does.